crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/prosecution_concluded/validator/ProsecutionConcludedValidator.java

```diff
@@ -22,12 +39,37 @@ public void validateRequestObject(ProsecutionConcluded prosecutionConcluded) {
             throw new ValidationException(PAYLOAD_IS_NOT_AVAILABLE_OR_NULL);
     }
 
-    public Optional<Void> validateOuCode(String ouCode) {
-
-        if (!isBlank(ouCode)) {
-            return Optional.empty();
-        } else {
+    public String validateOuCode(String ouCode) {
+        if (isBlank(ouCode)) {
             throw new ValidationException(OU_CODE_IS_MISSING);
         }
+
+        try {
+            return crownCourtCodeHelper.getCode(ouCode);
+        }
+        catch (MAATCourtDataException ex) {
+            throw new ValidationException(OU_CODE_LOOKUP_FAILED);
+        }
+    }
+
+    public void validateMaatId(String message) {
+        if (StringUtils.isEmpty(message)) {
+            throw new ValidationException(MAAT_ID_NOT_SUPPLIED);
+        }
+
+        try {
+            JsonObject msgObject = JsonParser.parseString(message).getAsJsonObject();
+            msgObject.get("maatId").getAsInt();
```

Comment:
I would avoid doing this in the validator, we're already deserialising the message in the listener where this is called from, so I would just rejig the calls and simplify this down.

Reply:
Are you saying that you would move the validation of the MAAT ID into the listener then, or perhaps pass the result of `msgObject.get("maatId")` into the validator? The reason I've done it in the validator is to group the validation logic.

Reply:
Yes, so my thought was to just pass in the MAAT_ID. Right now we're deserialising the message to retrieve the MAAT_ID in the validator and then doing it again a few lines afterwards in the listener, if we just swap around the calls we can pass in the MAAT_ID.

```java
prosecutionConcludedValidator.validateMaatId(message);
queueMessageLogService.createLog(MessageType.PROSECUTION_CONCLUDED, message);
ProsecutionConcluded prosecutionConcluded = gson.fromJson(message, ProsecutionConcluded.class);
```

Reply:
I guess the way we could do it is run `ProsecutionConcluded prosecutionConcluded = gson.fromJson(message, ProsecutionConcluded.class);` first, then basically do what we're doing in the ProsecutionConcludedValidator here instead, because if the MAAT ID is something invalid like `A123456` then we'll get an exception when calling `gson.fromJson(message, ProsecutionConcluded.class)` because the type will not match what's expected in ProsecutionConcluded. This would avoid us needing to deserialise twice, but then effectively moves the logic from `validateMaatId` into the listener which spreads the validation logic further around.

Reply:
As discussed, keeping this logic separate because of the possibility of a type conversion exception.